lib/resolveurl/plugins/streamrapid.py

```diff
@@ -16,38 +16,38 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
 
+import re
+import json
+import base64
 from six.moves import urllib_parse
 from resolveurl.plugins.lib import helpers
 from resolveurl import common
 from resolveurl.resolver import ResolveUrl, ResolverError
-import re
-import json
 
 
 class StreamRapidResolver(ResolveUrl):
     name = "streamrapid"
-    domains = ['streamrapid.ru']
-    pattern = r'(?://|\.)(streamrapid\.ru)/embed-([^\n]+)'
+    domains = ['streamrapid.ru', 'rabbitstream.net']
+    pattern = r'(?://|\.)((?:rabbitstream|streamrapid)\.(?:ru|net))/embed-([^\n]+)'
 
     def get_media_url(self, host, media_id):
         if '$$' in media_id:
             media_id, referer = media_id.split('$$')
             referer = urllib_parse.urljoin(referer, '/')
         else:
-            referer = False
+            # Needs to be hard coded for now if nothing is passed in.
+            referer = 'https://streamrapid.ru/'
         web_url = self.get_url(host, media_id)
         rurl = urllib_parse.urljoin(web_url, '/')
-        if not referer:
-            referer = rurl
-        domain = 'aHR0cHM6Ly9zdHJlYW1yYXBpZC5ydTo0NDM.'
         headers = {'User-Agent': common.FF_USER_AGENT,
                    'Referer': referer}
```

Comment:
Restore line 37, remove lines 38, 39
Do not remove line 40 and 41
The logic was correct before your changes, that is it will use user provided referer if not will use rurl as referer

Reply:
Have you tested that with any new embeds with the new domain? rurl doesn't work with new domain on the first request and that is why I changed it to the old one.

Reply:
A user can still pass in a referer if needed. Only thing i really changed there is....
If nothing is passed use the streamrapid domain vs new domain as the referer, that is all.

Reply:
yeah I didnt test the new domain as I didnt have embeds. okay, will merge, Thanks

Reply:
Cool no problem, thank you. I believe this is their new domain, based off of the main source i pull these from 😉